integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/message/processor/test/failover/FailoverForwardingProcessorTestCase.java

```diff
@@ -0,0 +1,89 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.message.processor.test.failover;
+
+import org.wso2.carbon.integration.common.admin.client.LogViewerClient;
+import org.wso2.carbon.logging.view.stub.types.carbon.LogEvent;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.esb.integration.common.utils.Utils;
+import org.wso2.esb.integration.common.utils.clients.axis2client.AxisServiceClient;
+import org.wso2.esb.integration.common.utils.servers.ActiveMQServer;
+
+import static org.testng.Assert.assertTrue;
+
+/**
+ * This test case tests whether failover forwarding of messages happen
+ * with the failover processor.
+ */
+public class FailoverForwardingProcessorTestCase extends ESBIntegrationTest{
+    private static LogViewerClient logViewer;
+    AxisServiceClient client = new AxisServiceClient();
+    private ActiveMQServer activeMQServer = new ActiveMQServer();
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        activeMQServer.startJMSBroker();
+        super.init();
+        logViewer = new LogViewerClient(context.getContextUrls().getBackEndUrl(), sessionCookie);
+    }
+
+    @Test(groups = "wso2.esb",
+          description = "Testing guaranteed delivery scenario")
+    public void testFailover() throws Exception {
+        loadESBConfigurationFromClasspath("/artifacts/ESB/synapseconfig/failover/failoverTest.xml");
+        logViewer.clearLogs();
+        client.sendRobust(Utils.getStockQuoteRequest("RMQ"), getProxyServiceURLHttp("failoverTestProxy"),
+                "getQuote");
+        Thread.sleep(10000);
+        assertTrue(isMatchFound("RMQ Company"),
+                "Message not dispatched from the store and sent to the backend!");
+
+        activeMQServer.stopJMSBroker();
+        logViewer.clearLogs();
+        client.sendRobust(Utils.getStockQuoteRequest("RMQ"), getProxyServiceURLHttp("failoverTestProxy"),
+                "getQuote");
+
+        activeMQServer.startJMSBroker();
+        Thread.sleep(10000);
+        assertTrue(isMatchFound("RMQ Company"),
+                "Message not dispatched from the store and sent to the backend!");
+    }
+
+    private boolean isMatchFound(String matchStr) throws Exception {
+        boolean isSet = false;
+        LogEvent[] logs = logViewer.getAllSystemLogs();
+        int size = logs.length;
+        for (int i = size-1; i >= 0; i--) {
+            if (logs[i].getMessage().contains(matchStr)) {
+                isSet = true;
```

Comment:
can simply `return true;`
instead of setting isSet and using break

Reply:
fixed